exercicios/para-casa/sistema_de_alunas.py

```diff
@@ -36,24 +36,91 @@ def obter_opcao():
         return codigo_opcao
     
 def incluir_nova_aluna():
-    pass
-    #TODO - Implentar a função
+    nome = input("Insira o nome:\n")
+    sobrenome = input("Insira o sobrenome:\n")
+    turma = input("Insira a turma:\n")
+
+    notas_input = input("Insira as notas separadas por vírgula:\n")
+    notas = notas_input.split(",")
+
+    presenca_input = input("Insira as presenças separadas por vírgula (1 para presença, 0 para falta)")
+    presenca_lista = presenca_input.split(",")
+    presenca = [True if cada.strip() == "1" else False for cada in presenca_lista]   
+
+    participacao = float(input("Insira a nota de participação"))
+
+    if notas and all(0 <= len(nota) <= 10 for nota in notas):
+        global dataset
+        dataset[(nome, sobrenome)] = {
+            "Turma": turma,
+            "Notas": notas,
+            "Presença": presenca,
+            "Participação": participacao
+        }
+        print("Aluna adicionada com sucesso!")
+    else:
+        print("Verifique as notas.")
     
 def consultar_lista_alunas():
-    pass
-    #TODO - Implentar a função
+    if not dataset:
+        print("Não há alunas cadastradas.")
+    else:
+        for nome, sobrenome in dataset.keys():
+            print(f"{nome} {sobrenome}")
+
     
 def consultar_faltas_aluna():
-    pass
-    #TODO - Implentar a função
+    nome_completo = input("Nome completo da aluna: ")
+
+    aluna_encontrada = False
+    for key in dataset.keys():
+        if f"{key[0]} {key[1]}" == nome_completo:
+            faltas = dataset[key]["Presença"].count(False)
+            print(f"A aluna {nome_completo} possui {faltas} faltas.")
+            aluna_encontrada = True
+            break
```

Comment:
Temos uma complexidade desnecessária no código. O acesso aos dados de uma aluna é feita pela chave, então podemos só acessar os valores. Pensar em códigos mais simples é algo importante na programação.

**Sugestão:**
```suggestion
    nome = input('Informe o primeiro nome da aluna: ').title()
    sobrenome = input('Informe o sobrenome da aluna: ').title()
    chave = (nome, sobrenome)
    if chave in dataset:
        faltas = dataset[chave]["Presença"].count(False)
        print(f"A aluna {nome_completo} possui {faltas} faltas.")
```